src/credentials/FabricTable.cpp

```diff
@@ -520,70 +524,58 @@ void FabricTable::ReleaseFabricIndex(FabricIndex fabricIndex)
 
 FabricInfo * FabricTable::FindFabric(P256PublicKeySpan rootPubKey, FabricId fabricId)
 {
-    static_assert(kMaxValidFabricIndex <= UINT8_MAX, "Cannot create more fabrics than UINT8_MAX");
-    for (FabricIndex i = kMinValidFabricIndex; i <= kMaxValidFabricIndex; i++)
+    for (auto & fabric : mStates)
     {
-        FabricInfo * fabric = FindFabricWithIndex(i);
-        if (fabric == nullptr)
+        if (!fabric.IsInitialized())
         {
             continue;
         }
         P256PublicKeySpan candidatePubKey;
-        if (fabric->GetRootPubkey(candidatePubKey) != CHIP_NO_ERROR)
+        if (fabric.GetRootPubkey(candidatePubKey) != CHIP_NO_ERROR)
         {
             continue;
         }
-        if (rootPubKey.data_equal(candidatePubKey) && fabricId == fabric->GetFabricId())
+        if (rootPubKey.data_equal(candidatePubKey) && fabricId == fabric.GetFabricId())
         {
-            LoadFromStorage(fabric);
-            return fabric;
+            return &fabric;
         }
     }
     return nullptr;
 }
 
 FabricInfo * FabricTable::FindFabricWithIndex(FabricIndex fabricIndex)
 {
-    if (fabricIndex >= kMinValidFabricIndex && fabricIndex <= kMaxValidFabricIndex)
+    for (auto & fabric : mStates)
     {
-        FabricInfo * fabric = &mStates[fabricIndex - kMinValidFabricIndex];
-        LoadFromStorage(fabric);
-        return fabric;
+        if (!fabric.IsInitialized())
+        {
+            continue;
+        }
+
+        if (fabric.GetFabricIndex() == fabricIndex)
+        {
+            return &fabric;
+        }
     }
 
     return nullptr;
 }
 
 FabricInfo * FabricTable::FindFabricWithCompressedId(CompressedFabricId fabricId)
```

Comment:
Where is this used? We have to return `nullptr` if there are two matching ones. Could you "find first" and then continue the loop, and if a count reaches > 1 match, we error with nullptr?

Reply:
It's used in `OperationalDeviceProxy`.

I agree that this is not a good API, and we need to stop using compressed ids for any sort of identification of things.  Once https://github.com/project-chip/connectedhomeip/pull/16000 merges we can work on fixing these things.